app/views/devise/registrations/_new.html.erb

```diff
@@ -28,13 +28,27 @@
   <% end %>
 
   <fieldset>
-    <%= f.text_field :password, autocomplete: "off", id: "signup_password", placeholder: "Your Password *" %>
+    <%= f.password_field :password, autocomplete: "off", id: "signup_password", placeholder: "Your Password *" %>
   </fieldset>
 
   <fieldset>
-    <%= f.text_field :password_confirmation, autocomplete: "off", placeholder: "Re-enter Your Password *" %>
+    <%= f.password_field :password_confirmation, autocomplete: "off", placeholder: "Re-enter Your Password *" %>
+  </fieldset>
+
+  <fieldset>
+    <%= check_box_tag('hide_signup_pass', false, true) %>
+    <%= label_tag('hide_signup_pass', 'Show password', class: 'plain') %>
   </fieldset>
 
   <%= f.submit "Sign Up" %>
 
 <% end %>
+
+<script type="text/javascript">
```

Comment:
@alexbrinkman My thoughts on putting this inline here is because it acts on a IDs that should only be in this partial.

I could put this into its own JS partial to include on the main login page, outside from the form partials, or I could make two partials, one for each form type (login and signup), to include in their respective partials.

I guess my thoughts on why I included the work inline is that it's rather localized to just these partials, both partials are on the same page (so, I can't use a single function that operates on a generic "password_field" ID that's reused throughout the whole app), and I feel that making a partial for each login/signup form partial, only to include each in it's respective form partial, adds a layer of abstraction without any significant benefit.

If we used the exact same password field ID throughout the entire app and I could reuse the partial everywhere it's needed to remove duplication, it would make sense. However, if this is just a style practice that we try to converge on, I'm ok sticking to it, even if I disagree.